packages/rs-drive-abci/src/abci/handler/mod.rs

```diff
@@ -208,33 +218,41 @@ where
 
         let block_execution_outcome::v0::BlockExecutionOutcome {
             app_hash,
-            tx_results,
+            state_transition_results,
             validator_set_update,
         } = run_result.into_data().map_err(Error::Protocol)?;
 
         // We need to let Tenderdash know about the transactions we should remove from execution
-        let (tx_results, tx_records): (Vec<ExecTxResult>, Vec<TxRecord>) = tx_results
+        let tx_results_and_records = state_transition_results
             .into_iter()
             .map(|(tx, result)| {
-                if result.code > 0 {
+                let tx_result = result.try_into_tx_result(platform_version)?;
```

Comment:
there's a try_into_platform_versioned that should be used instead.

Reply:
Didn't know that we have it. Nice!